app/lib/oauth_pre_authorization_extension.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module OauthPreAuthorizationExtension
+  extend ActiveSupport::Concern
+
+  included do
+    validate :code_challenge_method_s256, error: Doorkeeper::Errors::InvalidCodeChallengeMethod
+  end
+
+  def validate_code_challenge_method_s256
+    code_challenge.blank? || (code_challenge_method.present? && code_challenge_method == 'S256')
```

Comment:
I think we can just write this as:

```suggestion
    code_challenge.blank? || code_challenge_method == 'S256'
```

Reply:
Yes, I agree.